dhall-json/src/Dhall/JSON.hs

```diff
@@ -374,19 +370,35 @@ dhallToJSON e0 = loop (Core.alphaNormalize (Core.normalize e0))
                  ,  (   "nesting"
                     ,   nesting
                     )
-                 ] | isInlineNesting nesting -> do
-                    let contents' =
-                            Dhall.Map.insert
-                                field
-                                (Core.TextLit
-                                    (Core.Chunks
-                                        []
-                                        alternativeName
-                                    )
-                                )
-                                contents
+                 ] | isInlineNesting nesting
+                   , Just (alternativeName, Just (Core.RecordLit kvs)) <- getContents contents -> do
```

Comment:
I wonder whether we should report an error in this case:
```suggestion
                   , Just (alternativeName, Just notARecordLit) <- getContents contents -> do
```

Non-record alternatives are supported with `Nested`, so users might also try them with `Inline`.

Reply:
@sjakobi: Yeah, we probably should report an error here.  It's up to you if you want to include that in this pull request or not

Reply:
I'll add the error in a follow-up PR.